cpp/src/strings/convert/convert_durations.cu

```diff
@@ -316,6 +325,48 @@ struct duration_to_string_fn : public duration_to_string_size_fn<T> {
     return ptr + digits + 1;
   }
 
+  inline __device__ char* nanosecond(char* ptr, duration_component const* timeparts)
+  {
+    auto value = timeparts->subsecond;
+    *ptr       = '.';
+    for (int idx = 9; idx > 0; idx--) {
+      *(ptr + idx) = '0' + std::abs(value % 10);
+      value /= 10;
+    }
+    return ptr + 10;
+  }
+
+  inline __device__ char* pandas_format(char* ptr, duration_component const* timeparts)
```

Comment:
This seems only specific to the csv writer. What do you think about moving this format code to an `io/csv/` source file instead?

Reply:
The reason for keeping it here is, this could be helpful for converting duration column to string column or printing duration column too.
Also csv writer calls `from_<dtype>()` functions for most of dtypes. If moved to io/csv/, most of `from_durations()` code should be duplicated.

Reply:
I guess I'm not seeing that much duplication. The format is hardcoded to "%P" so there is no real need to parse the format. And handling of the `%P` is completely handled by this function. The only reuse here is the `dissect_duration` which could be moved into a common header.
I suppose you picked the `%P` to mean "Pandas" but the intention here was the format specifiers matched closely to the C/C++ function [strptime](https://man7.org/linux/man-pages/man3/strptime.3.html) and I fear that `%P` may be unexpected (or used in the future). Also, I think it reasonable to not support printing in `%P` format since it is not part of the `strptime` specifiers.

Reply:
Removed